chaosgcp/apphub/actions.py

```diff
@@ -0,0 +1,270 @@
+    """
+    client = compute_v1.UrlMapsClient()
+    request = compute_v1.AggregatedListUrlMapsRequest(project=project_id)
+
+    for response in client.aggregated_list(request=request):
+        scope, url_maps_scoped_list = response
+        region = scope.split("/")[-1] if "/" in scope else "global"
+
+        if url_maps_scoped_list.url_maps:
+            for url_map in url_maps_scoped_list.url_maps:
+                yield {
+                    "region": region,
+                    "service": url_map.default_service[
+                        url_map.default_service.find("/projects/") :
+                    ]
+                    if "/projects/" in url_map.default_service
+                    else url_map.default_service,
+                    "self_link": url_map.self_link,
+                }
+
+
+def get_urlMap_self_links_for_matching_services(
+    service_uris, url_map_essentials
+):
+    """Finds self_links of URL maps whose services match the given service URIs."""
+    matching_self_links = []
+    logger.info(
+        "Finds self_links of URL maps whose services match with given services of Application"
+    )
+    for details in url_map_essentials:
+        if details["service"] in service_uris:
+            matching_self_links.append(details["self_link"])
+            logger.info(details["self_link"])
+    return matching_self_links
+
+
+def url_map_exists(url_map_value: str, project_id: str, region: str, application_name: str):
+    """Checks if a specific URL map value exists within AppHub-related URL maps.
+
+    This function first retrieves the selfLink URLs of all URL maps associated with services
+    in the specified AppHub application. It then searches these URL maps to determine if the
+    given `url_map_value` is present after the "urlMaps/" segment.
+
+    Args:
+        url_map_value (str): The specific value (e.g., "my-url-map") to search for after "urlMaps/" in the URL.
+        project_id (str): The ID of the Google Cloud project containing the AppHub application.
+        region (str): The region where the AppHub application is deployed.
+        application_name (str): The name of the AppHub application.
+
+    Returns:
+        bool: True if the `url_map_value` is found within the URL maps associated with
+              the specified AppHub application, False otherwise.
+    """
+    try:
+        availableUrlMaps = get_urlMap_self_links_for_matching_services(
+            apphub_app_list_services(project_id, region, application_name),
+            get_url_map_essentials(project_id),
+        )
+        for umap in availableUrlMaps:
+            index = umap.find("/urlMaps/")
+            if (
+                index != -1
+                and umap[index + len("/urlMaps/") :] == url_map_value
+            ):
+                logger.info(
+                    "given url_map is found within the URL maps associated with the specified AppHub application"
+                )
```

Comment:
I'm not in favour to have info level for statements like this one. Info is really about the experiment mechanic, not extension debug traces.